### PR TITLE
feat: add extension rust-document

### PR DESCRIPTION
**Pull request by waynexia**
Signed-off-by: Ruihang Xia <waynestxia@gmail.com>

## Description

Search rust crates' document easily

## Screencast

screenshot 2023-02-08 19 00 47
screenshot 2023-02-08 19 01 57

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/rust-document/package.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "rust-document",
```
Comment (pernielsentikaer) [inline extensions/rust-document/package.json:3]:
```suggestion
  "name": "rust-documentation",
```

File: extensions/rust-document/package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  "title": "Rust Document",
```
Comment (pernielsentikaer) [inline extensions/rust-document/package.json:4]:
```suggestion
  "title": "Rust Documentation",
```

File: extensions/rust-document/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  "description": "Search Rust document quickly",
```
Comment (pernielsentikaer) [inline extensions/rust-document/package.json:5]:
```suggestion
  "description": "Search Rust documentation quickly",
```

File: extensions/rust-document/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  "categories": [
+    "Developer Tools",
+    "Documentation",
+    "Productivity"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "rs",
```
Comment (pernielsentikaer) [inline extensions/rust-document/package.json:17]:
```suggestion
      "title": "Search Documentation",
      "subtitle": "Rust",
```

File: extensions/rust-document/package.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      "description": "Search Rust document quickly",
```
Comment (pernielsentikaer) [inline extensions/rust-document/package.json:18]:
```suggestion
      "description": "Search Rust documentation quickly",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

Some suggestions might be wrong if it's not the Rust documentation, but let's see what we can improve 😊

* Some results are not working (Seems to be all with 0.0.0 as version)
 
Raycast-v5QnWDvb@2x


I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by waynexia** [comment]
Hi @pernielsentikaer, thanks for your review. I've added a filter to opt-out crates with version "0.0.0", now it looks like this: 
截屏2023-02-12 01 16 36


By the way, just curious, is there any plan or discussion to accept extensions to be added as submodules?

**Comment by waynexia** [comment]
Could you please help me to fix the CI? I've run `eslint --init` and `eslint --fix` in my env and no change generated.

**Comment by pernielsentikaer** [comment]
You should see the same errors when running `npm run lint` or checking [this link](https://github.com/raycast/extensions/actions/runs/4152376325/jobs/7183291695)

**Comment by waynexia** [comment]
Got a few busy weekends, I would resolve those lints soon

**Pull request closed without merging**